File: cpp/724PivotIndex.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
      int n = nums.size();
      vector<int> l, r;
      int s = 0;
      int currentLeftSum = 0;
      for(auto x:nums){
          s+=x;
          l.push_back(currentLeftSum);
          currentLeftSum+=x;
          r.push_back(-currentLeftSum);
      }
      for(int i = 0; i< n; i++){
          if(l[i]==(r[i]+s))
            return i;
      }
      return -1;
    }
};

File: python/724PivotIndex.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        l,r,ls,rs,n=[],[],0,0,len(nums)
        for i in range(n):
            l.append(ls)
            ls+=nums[i]
            r.append(rs)
            rs+=nums[n-i-1]
        r.reverse()
        for i in range(n):
            if r[i] == l[i]:
                return i
        return -1
